src/ir_passes/lower/utils.rs

```diff
@@ -78,11 +79,39 @@ pub(super) fn port_name(
     }
 }
 
-/// Returns the name of an [ir::Component]
+/// Returns the name of an [Component]
 pub(super) fn comp_name(idx: CompIdx, ctx: &impl Ctx<Component>) -> String {
     ctx.get(idx)
         .src_info
         .as_ref()
         .map(|src| src.name.to_string())
         .unwrap_or_else(|| format!("comp{}", idx.get()))
 }
+
+/// Calculates the max states used for every fsm.
+pub fn max_states(idx: CompIdx, ctx: &Context) -> LinkedHashMap<EventIdx, u64> {
```

Comment:
Why is the key `EventIdx` and not `Foreign<EventIdx>` or something? Since this computation happens over every component, we're going to end up overwriting the wrong `EventIdx`

Reply:
Oh, my bad, I guess it's computing it for all the events in a specific component. The `ctx` parameter confused me.

Reply:
This function only calculates max states for one component, becuase dump interface only requires it to be calculated for main, and its calculated on demand in lower as well.

Reply:
Can we just pass the `ir::Component` by reference to this method instead of passing in the context?